modules/virtual_network_gateway/variables.tf

```diff
@@ -136,59 +129,95 @@ variable "ip_configuration" {
       subnet_id        = "ID_for_subnet_GatewaySubnet"
     }
   ]
-
+  ```
   EOF
-  type        = list(any)
+  default     = []
+  nullable    = false
+  type = list(object({
+    name                          = optional(string, "vnetGatewayConfig")
+    create_public_ip              = bool
+    public_ip_name                = optional(string)
```

Comment:
when looking at the code it looks like the name property is required, not optional, see `main.tf`, lines `101`&`108` and `118`&`120`

Reply:
yes, you are right, fixed